include/pybind11/pybind11.h

```diff
@@ -561,13 +571,34 @@ class cpp_function : public function {
                 // We'll need to copy this if we steal some kwargs for defaults
                 dict kwargs = reinterpret_borrow<dict>(kwargs_in);
 
+                // 1.5. Fill in any missing pos_only args from defaults if they exist
+                if (args_copied < func.nargs_pos_only) {
+                    for (; args_copied < func.nargs_pos_only; ++args_copied) {
+                        const auto &arg = func.args[args_copied];
+                        handle value;
+
+                        if (arg.value) {
+                            value = arg.value;
+                        }
+                        if (value) {
+                            call.args.push_back(value);
+                            call.args_convert.push_back(arg.convert);
+                        } else
+                            break;
```

Comment:
So basically, if I understand correctly, the implementation is so short/simple because `pos_only` works by failing early, before keyword args are even considered?

Interesting to link to #2284. But further than that, I guess it doesn't really matter, because pybind11 doesn't generate error messages here, so there's no reason to detect a pos-only argument was passed as keyword.

Reply:
Yes. kw_only uses basically the same trick, as well.